components/esp-tls/esp_tls.c

```diff
@@ -308,17 +313,20 @@ static esp_err_t esp_tls_set_socket_non_blocking(int fd, bool non_blocking)
 
     if (fcntl(fd, F_SETFL, flags) < 0) {
         str_buf_t err_desc = esp_err_to_name_with_alloc_str_buf(errno);
-        ESP_LOGE(TAG, "[sock=%d] set blocking/nonblocking error: %d (%s)", fd, errno,
-                 (NULL != err_desc.buf) ? err_desc.buf : "");
+        ESP_TLS_LOGE_FUNC("[sock=%d] set blocking/nonblocking error: %d (%s)",
+                 fd, errno, (NULL != err_desc.buf) ? err_desc.buf : "");
         str_buf_free_buf(&err_desc);
         return ESP_ERR_ESP_TLS_SOCKET_SETOPT_FAILED;
     }
     return ESP_OK;
 }
 
-static inline esp_err_t esp_tls_tcp_connect(const ip_addr_t* const p_remote_ip, int port, const esp_tls_cfg_t *cfg,
+static esp_err_t esp_tls_tcp_connect(const ip_addr_t* const p_remote_ip, int port, const esp_tls_cfg_t *cfg,
             esp_tls_error_handle_t error_handle, int *sockfd)
 {
+    char remote_ip_str[IP4ADDR_STRLEN_MAX];
+    ip4addr_ntoa_r(&p_remote_ip->u_addr.ip4, remote_ip_str, IP4ADDR_STRLEN_MAX);
```

Comment:
Is this safe on IPv6 only servers? 

It seems to me that `ip4addr_ntoa_r` calls `ip4_addr_get_u32` which dereferences given argument. Since the argument is the address of `p_remote_ip->u_addr.ip4` this *seems* like even in case there is no ipv4 address available, it would just point to some uninitialized memory in a worst case and it would be used as a readonly.

Reply:
> Is this safe on IPv6 only servers?
> 
> It seems to me that `ip4addr_ntoa_r` calls `ip4_addr_get_u32` which dereferences given argument. Since the argument is the address of `p_remote_ip->u_addr.ip4` this _seems_ like even in case there is no ipv4 address available, it would just point to some uninitialized memory in a worst case and it would be used as a readonly.

No, some patches are not IPv6 compatible. But IPv6 support is disabled in this project.
Do you think IPv6 support should be enabled?  Could this be planned for a future version 1.16.x, what do you think?

Reply:
Let's leave the IPv6 support for 1.16 or some other future support in this case, so far we have not had a user request it.